### PR TITLE
added batched string kernel and example notebook

### PR DESCRIPTION
**Pull request by aryandeshwal**
This PR adds the subsequence string kernel ([Ref 1](https://arxiv.org/pdf/2010.00979.pdf) [Ref 2](https://aclanthology.org/I17-2012/) [Ref 3](https://www.jmlr.org/papers/volume2/lodhi02a/lodhi02a.pdf)) implemented in GpyTorch. It is batch-vectorized and GPU-compatible and follows Henry's Tensorflow implementation from [here](https://github.com/henrymoss/BOSS/blob/master/boss/code/kernels/string/tf_string_kernel.py). I tested it on the Photoswitch dataset in the example notebook. Thank you!

### CODE REVIEW

File: gauche/kernels/string_kernels/sskkernel.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
+# GPU-compatible gpytorch version of the substring kernel implemented in https://github.com/henrymoss/BOSS/tree/master/boss/code/kernels/string
```
Comment (Ryan-Rhys) [inline gauche/kernels/string_kernels/sskkernel.py:1]: Should we format the module-level docstring as in: 

https://github.com/leojklarner/gauche/blob/main/gauche/kernels/gnn_kernels/pretrained_kernel.py

for consistency across the codebase?

File: gauche/kernels/string_kernels/sskkernel.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+# [1] Beck, D., & Cohn, T. Learning kernels over strings using Gaussian processes. In Proceedings of the Eighth International Joint Conference on Natural Language Processing (Volume 2: Short Papers) (pp. 67-73).
+# or appendix of [2] Moss, H., Leslie, D., Beck, D., Gonzalez, J., & Rayson, P. (2020). Boss: Bayesian optimization over string spaces. Advances in neural information processing systems, 33, 15476-15486.
+
+import math
+import itertools
+import torch
+from gpytorch.kernels import Kernel
+from gpytorch.constraints import Interval
+
+def pad(s, length):
```
Comment (Ryan-Rhys) [inline gauche/kernels/string_kernels/sskkernel.py:12]: Should we format function docstrings as in:

https://github.com/leojklarner/gauche/blob/main/gauche/kernels/gnn_kernels/pretrained_kernel.py

for consistency across the codebase?

**Comment by Ryan-Rhys** [comment]
Looks great! Fantastic stuff! Happy to merge after nits.

File: gauche/kernels/string_kernels/sskkernel.py (hunk 1 above)
Comment (aryandeshwal) [reply gauche/kernels/string_kernels/sskkernel.py]: updated both the top-level and function-level docstring with the new style

**Pull request merged**